src/sentry/rules/actions/utils.py

```diff
@@ -34,14 +34,37 @@ def generate_diff_labels(
     present_state: dict[str, Any],
     prior_state: dict[str, Any],
     rule: Rule,
-    changed_data: DefaultDict[str, list[str]],
     key: str,
-    statement: str,
 ) -> DefaultDict[str, list[str]]:
-    for data in prior_state.get(key, []):
-        if data not in present_state.get(key, []):
+    added_statement = "Added {} '{}'"
+    removed_statement = "Removed {} '{}'"
+    added_items = []
+    changed_data: DefaultDict[str, list[str]] = defaultdict(list)
+    # Added items that are not seen in the prior rule data
+    for data in prior_state:
+        if data not in present_state:
+            label = generate_rule_label(rule.project, rule, data)
+            changed_data[data["id"]].append(added_statement.format(key, label))
+            added_items.append(data["id"])
+
+    # Check if the id is in both but the data has changed
+    for item in added_items:
+        if changed_data.get(item):
+            for data in prior_state:
+                for present_data in present_state:
+                    if data["id"] == item and present_data["id"] == item:
+                        if present_data != data:
```

Comment:
![huh](https://github.com/getsentry/sentry/assets/67301797/cdfaa140-85d5-4227-8af3-8eda7bfda9fc)

Reply:
😅 it's hard to compare nested data